chain/client/src/types.rs

```diff
@@ -264,6 +266,29 @@ impl Message for Status {
     type Result = Result<StatusResponse, String>;
 }
 
+pub struct GetNetworkInfo {}
+
+impl Message for GetNetworkInfo {
+    type Result = Result<NetworkInfoResponse, String>;
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct PeerInfo {
```

Comment:
With this one, there are three declaration of `PeerInfo` through the code. We should keep one declaration in `primitives` only.

Reply:
Good point! I didn't see the PeerInfo in primitives. If it's only in primitives and used in chain/netowork and chain/client that's great.

Reply:
Should we only keep it in network? Ideally keep primitives as small as possible

Reply:
Sounds good. In a quick find, I do not see any usage of PeerInfo etc in primitives. Probably it has been moved to network & upper layer and definition in primitives are outdated. Will confirm then refactor

Reply:
Fixed, primitive/network.rs isn't used anywhere, removed. network/types.rs PeerInfo is now used